src/core_modules/capture-core/metaDataStoreLoaders/programs/quickStoreOperations/storeProgramRules.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import { quickStoreRecursively } from '../../IOUtils';
+import { getContext } from '../../context';
+
+const convert = (() => {
+    const getProgramRuleActions = (apiProgramRuleActions) => {
+        const getClearProps = () => ({
+            programStageSection: undefined,
+            programStage: undefined,
+            dataElement: undefined,
+            trackedEntityAttribute: undefined,
+            optionGroup: undefined,
+            option: undefined,
+        });
+
+        const getProgramStageSectionId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.programStageSection && d2ProgramRuleAction.programStageSection.id;
+        const getProgramStageId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.programStage && d2ProgramRuleAction.programStage.id;
+        const getDataElementId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.dataElement && d2ProgramRuleAction.dataElement.id;
+        const getTrackedEntityAttributeId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.trackedEntityAttribute && d2ProgramRuleAction.trackedEntityAttribute.id;
+        const getOptionGroupId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.optionGroup && d2ProgramRuleAction.optionGroup.id;
+        const getOptionId = d2ProgramRuleAction =>
+            d2ProgramRuleAction.option && d2ProgramRuleAction.option.id;
+
+        return apiProgramRuleActions.map(apiProgramRuleAction => ({
+            ...apiProgramRuleAction,
+            ...getClearProps(),
+            programStageSectionId: getProgramStageSectionId(apiProgramRuleAction),
+            programStageId: getProgramStageId(apiProgramRuleAction),
+            dataElementId: getDataElementId(apiProgramRuleAction),
+            trackedEntityAttributeId: getTrackedEntityAttributeId(apiProgramRuleAction),
+            optionGroupId: getOptionGroupId(apiProgramRuleAction),
+            optionId: getOptionId(apiProgramRuleAction),
+        }));
+    };
+
+    return (response) => {
+        const apiProgramRules = (response && response.programRules) || [];
+
+        return apiProgramRules
+            .map(apiProgramRule => ({
+                ...apiProgramRule,
+                program: undefined,
+                programStage: undefined,
```

Comment:
Added a comment to the code.

Reply:
Thanks for adding the comment. The comment doesnt explain why these values are `undefined`. 

Again this is the strange thing for me here and this is why asked a comment so that we could explain better the reasons behind having to pass `undefined`.

Reply:
Clarified comment 🙂

Reply:
🙇